src/fractionatedmorse.rs

```diff
@@ -0,0 +1,333 @@
+    /// characters. Furthermore, it is possible that a purely alphabetic message will not produce
+    /// valid Morse code, in which case an `Err` will again be returned.
+    ///
+    /// An additional `i` or `e` may be present at the end of the decrypted message due to padding
+    /// added during the encryption process.
+    ///
+    /// # Examples
+    /// Basic usage:
+    ///
+    /// ```
+    /// use cipher_crypt::{Cipher, FractionatedMorse};
+    ///
+    /// let fm = FractionatedMorse::new(String::from("key")).unwrap();
+    /// assert_eq!("attack at dawn!", fm.decrypt("cpsujixpzycbuuiq").unwrap());
+    /// ```
+    fn decrypt(&self, cipher_text: &str) -> Result<String, &'static str> {
+        // Decryption process:
+        //   (1) The keyed alphabet is obtained from the key.
+        //   (2) Each ciphertext char is located by index in the keyed alphabet.
+        //   (3) The indices are convert to 3 digit ternary and the substitution '.' for 0,
+        //       '-' for 1 and '|' for 2 is made to produce a trigraph for each letter.
+        //   (4) These trigraphs are substituted for each letter in the message and concatenated
+        //       to produce a Morse string.
+        //   (5) The Morse message is decoded.
+        //
+        // Example: Key: `alphabet`, Ciphertext: `atsphcmr`
+        //   (1) The alphabet `alphbetcdfgijkmnoqrsuvwxyz` is produced.
+        //   (2) a(0), t(6), s(19), p(2), h(3), c(7), m(14), r(18)
+        //   (3) 0  -> 000 ->  ...
+        //       6  -> 020 ->  .|.
+        //       19 -> 201 ->  |.-
+        //       2  -> 002 ->  ..|
+        //       and so on.
+        //   (4) The Morse message `....|.|.-..|.-..|---||..` is produced.
+        //   (5) The plaintext `hello i` is recovered.
+        let frac_morse = FractionatedMorse::encrypt_frac_morse(&self.keyed_alphabet, cipher_text.to_string())?;
+        let plaintext = FractionatedMorse::decrypt_morse(frac_morse)?;
+        Ok(plaintext)
+    }
+}
+
+
+impl FractionatedMorse {
+
+    /// Takes a string and trys to convert it to Morse code, using the character `|` as a
+    /// separator. The Morse code is ended with two separators `||`. This function returns `Err`
+    /// if an unsupported symbol is present. The support characters are `a-z`, `A-Z`, `0-9`, spaces
+    /// and the special characters `@ ( ) . , : ' " ! ? - ; =`.
+    fn encrypt_morse(message: String) -> Result<String, &'static str> {
+        let mut morse = String::new();
+
+        // Convert each letter in message to corresponding Morse characters.
+        // We cannot have multiple spaces in a row, otherwise the invalid Fractionated Morse
+        // character `|||` will be produced. Therefore, split by whitespace and only add one space
+        // in-between each word.
+        for word in message.split_whitespace() {
+            for c in word.chars() {
+                if let Some(pos) = MORSE_ALPHABET.iter().position(|m| m.0 == c.to_ascii_lowercase()) {

```

Comment:
I did consider a hashmap, but it requires an external crate to get a macro for creating one (the equivilant of vec!). Seemed a bit extreme, but it would be handy.